code/iaas/logic-common/src/main/java/io/cattle/platform/network/impl/NetworkServiceImpl.java

```diff
@@ -103,20 +103,21 @@ public void releaseIpAddress(Network network, Object owner) {
     public String getNetworkMode(Map<String, Object> instanceData) {
         Map<String, Object> labels = CollectionUtils.toMap(instanceData.get(InstanceConstants.FIELD_LABELS));
         String mode = ObjectUtils.toString(labels.get(SystemLabels.LABEL_CNI_NETWORK));
-
-        if (mode == null && "true".equals(labels.get(SystemLabels.LABEL_RANCHER_NETWORK))) {
-            mode = NetworkConstants.NETWORK_MODE_MANAGED;
-        }
-
-        if (mode == null) {
-            if (instanceData.containsKey(DockerInstanceConstants.FIELD_NETWORK_MODE)) {
-                mode = ObjectUtils.toString(instanceData.get(DockerInstanceConstants.FIELD_NETWORK_MODE));
-            } else {
+        
+        if(mode == null) {
+            String dataMode = ObjectUtils.toString(instanceData.get(DockerInstanceConstants.FIELD_NETWORK_MODE));
+            if (inAllowedModes(dataMode) && "true".equals(labels.get(SystemLabels.LABEL_RANCHER_NETWORK))) {
                 mode = NetworkConstants.NETWORK_MODE_MANAGED;
+            } else {
+                mode = dataMode;
```

Comment:
are we fine assigning mode=dataMode if dataMode when !inAllowedModes? it falls under this else condition